WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m

```diff
@@ -138,11 +138,23 @@ - (void)loadDataWithOptions:(WPMediaLoadOptions)options success:(WPMediaSuccessB
     }
     // try to sync from the server
     MediaCoordinator *mediaCoordinator = [MediaCoordinator shared];
-    [mediaCoordinator syncMediaFor:self.blog success:^{
-        if (!localResultsAvailable && successBlock) {
-            successBlock();
-        }
-    } failure:failureBlock];
+
+    __weak __typeof(self) weakSelf = self;
+    [mediaCoordinator syncMediaFor:self.blog
+                           success:^{
+                               if (!localResultsAvailable && successBlock) {
+                                   successBlock();
+                               }
+                           } failure:^(NSError * _Nonnull error) {
+                               if ([error.domain isEqualToString:NSURLErrorDomain] && weakSelf.ignoreNetworkLoadingErrors && successBlock) {
```

Comment:
two questions:
1. Do we need to check the error domain here? or can we handle the same for any error?
2. Why are we checking the successBlock here?

Reply:
I was being cautious here — I think it might be slightly dangerous to outright ignore any errors this method can return — if there's a failure when parsing the response or a Core Data error, I don't think we want to swallow them the same way. 

I'm open to changing this and being more liberal if you think it's fine though :)

Reply:
Will we still see the annoying popup alert error on any of the other cases (like core data error). I believe we made a general design decision to not block the user with these annoying alerts on errors anymore unless the errors should block progress.
What do you think?

Reply:
Yeah, you're right. I changed it to handle any errors here.